vpr/src/route/annotate_routing.cpp

```diff
@@ -0,0 +1,68 @@
+/********************************************************************
+ * This file includes functions that are used to annotate routing results
+ * from VPR to OpenFPGA
+ *******************************************************************/
+/* Headers from vtrutil library */
+#include "vtr_assert.h"
+#include "vtr_time.h"
+#include "vtr_log.h"
+
+#include "vpr_error.h"
+#include "rr_graph.h"
+
+#include "annotate_routing.h"
+
+/********************************************************************
+ * Create a mapping between each rr_node and its mapped nets 
+ * based on VPR routing results
+ * - Store the net ids mapped to each routing resource nodes
+ * - Mapped nodes should have valid net ids (except SOURCE and SINK nodes)
+ * - Unmapped rr_node will use invalid ids 
+ *******************************************************************/
+vtr::vector<RRNodeId, ClusterNetId> annotate_rr_node_nets(const DeviceContext& device_ctx,
+                                                          const ClusteringContext& clustering_ctx,
+                                                          const RoutingContext& routing_ctx,
+                                                          const bool& verbose) {
+    size_t counter = 0;
+    vtr::ScopedStartFinishTimer timer("Annotating rr_node with routed nets");
+
+    vtr::vector<RRNodeId, ClusterNetId> rr_node_nets;
```

Comment:
I'm confused, how is this variable different than https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/master/vpr/src/base/vpr_context.h#L318 ?

Reply:
This is just a local variable created for fast look-up. I tried to use only the `trace` in routing context as @kmurray told me it is an official data structure to store routing results.

Reply:
Can you link to his comments?  I'm unclear if his comment matches your interpretation.

Regardless, if net_rr_terminals is not populated, you should just use the code for generating net_rr_terminals, which is here: https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/ed1a373d9c3409eea098612b6530f734c125ca03/vpr/src/route/route_common.cpp#L976

Reply:
https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1355#discussion_r438833451

Reply:
Ok, so the answer to my original question is that this data structure includes nodes other than the `SINK` and `SOURCE`.

Reply:
I understand the point here but the `net_rr_terminal` is not the fast look-up I am looking for. What I need is to find a cluster net id with a given `rr_node` id. The `net_rr_terminal` is a reversed mapping. Therefore, I used a straightforward mapping here.